### ISSUE TITLE
Reserve Enum with value 0

### ISSUE THREAD

**Issue by zrno** [protocolbuffers/protobuf#12307]
**Describe the problem you are trying to solve.**
Using the line `reserved 0;` in an enum is not possible. Reserving any other fields in an enum is possible.

**Comment by anandolee** [comment]
Yes the first enum value must be zero in proto3. Most projects that do not use 0 usually add an UNKNOWN = 0 to the enum
